framework/Optimizers/Optimizer.py

```diff
@@ -111,8 +111,12 @@ def getInputSpecification(cls):
     init.addSub(thresh)
     init.addSub(write)
     inputSpecification.addSub(init)
-
     # convergence
+
+    # central difference
+    cendiff = InputData.parameterInputFactory('centralDifference', contentType=InputData.BoolType)
+    gradhis = InputData.parameterInputFactory('useGradientHistory', contentType=InputData.BoolType)
```

Comment:
If you use the InputData, why  above you test for a string when you read/retrieve the variables?

Reply:
Understand...but the documentation on how to use the InputData came in 2017... The optimizer base classes in 2016... 

Anyhow, I understand that is not related to this algorithm, but if we never tackle the problem, the problem will stay forever. It is better to tackle it now (since we are not talking about 3 months of work) than later.

Reply:
I agree it should be done, but I think in the interest of smaller PRs being better, and each PR tackling finite issues, this should be done in a separete PR immediately following this one. I agree putting it off doesn't make sense, maybe we can just keep it distinct?